### PR TITLE
TGUI Robotics Console

### PR DESCRIPTION
**Pull request by gurfan**
vjWsbrPVzF
vjWsbrPVzF
vjWsbrPVzF

:cl:
 * imageadd: The robotics console now uses a fancy TGUI.

### CODE REVIEW

**Comment by Eneocho** [comment]
Does the console still work with TGUI disabled, like when you disable it for the GPS?

**Comment by gurfan** [comment]
uhhhhh

**Comment by Eneocho** [comment]
There, corrected the post. Also, you embedded the second image wrong.

**Comment by boy2mantwicethefam** [comment]
Meme UI but I'll take it

**Comment by DamianX** [comment]
Do you have any issue with tgui currently?

**Comment by gurfan** [comment]
With TGUI disabled in settings it still works (somehow), but you see the default BYOND bar at the top.

**Comment by DamianX** [comment]
Yeah. The "fancy tgui" option is supposed to do exactly that. It was abused for GPSs because people were having issues with tgui, so they could go back to nanoui. I have only got one (1) report of someone who cannot use tgui and it's probably something specific to their VM, so there's no reason to keep two UIs now.

Unrelated: the tgui guys have been nice enough to relicense everything, including some of the interfaces, to MIT so that we could use them. Would you mind doing the same with this one? This is assuming you wrote it. You can copy the comment I placed at the top of `Gps.js`.

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -175,7 +175,7 @@
 
 //Modules
 /obj/item/weapon/robot_module/standard
-	name = "standard robot module"
+	name = "Standard"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: WHY

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 2]
```diff
@@ -221,7 +221,7 @@
 	fix_modules()
 
 /obj/item/weapon/robot_module/medical
-	name = "medical robot module"
+	name = "Medical"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: WOULD

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 3]
```diff
@@ -293,7 +293,7 @@
 	fix_modules()
 
 /obj/item/weapon/robot_module/engineering
-	name = "engineering robot module"
+	name = "Engineering"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: YOU

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 4]
```diff
@@ -348,7 +348,7 @@
 	fix_modules()
 
 /obj/item/weapon/robot_module/security
-	name = "security robot module"
+	name = "Security"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: DO

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 5]
```diff
@@ -383,7 +383,7 @@
 	fix_modules()
 
 /obj/item/weapon/robot_module/janitor
-	name = "janitorial robot module"
+	name = "Janitor"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: THIS

File: code/game/machinery/computer/robot.dm [hunk 6]
```diff
@@ -41,228 +41,133 @@
...
+	data["cyborgs"] = list()
+	for(var/mob/living/silicon/robot/R in cyborg_list)
+		if(!can_control(R,user))
+			continue
+		var/list/cyborg_data = list(
+			name = R.name,
+			locked_down = R.lockdown,
+			status = R.stat,
+			charge = R.cell ? R.cell.percent() : null,
+			module = R.module ? "[R.module.name] Module" : "No Module Installed",
```
Comment (jknpj) [inline code/game/machinery/computer/robot.dm]: https://github.com/vgstation-coders/vgstation13/blob/ab73db104137216e99d51bc9e5398ed7957c5f33/__DEFINES/silicon.dm#L1-L48

Comment (jknpj) [reply code/game/machinery/computer/robot.dm]:
```suggestion
			module = R.module ? "[R.modtype] Module" : "No Module Installed",
```
Disregard defines and lists, acquire speed

**Pull request merged**